library/Translate/src/Juvix/Sexp/Structure.hs

```diff
@@ -702,3 +856,185 @@ toRecordNoPunned form
+  Sexp.list [Sexp.atom nameInfix, sexp1, sexp2, sexp3]
+
+----------------------------------------
+-- OpenIn
+----------------------------------------
+
+nameOpenIn :: NameSymbol.T
+nameOpenIn = ":open-in"
+
+isOpenIn :: Sexp.T -> Bool
+isOpenIn (Sexp.Cons form _) = Sexp.isAtomNamed form nameOpenIn
+isOpenIn _ = False
+
+toOpenIn :: Sexp.T -> Maybe OpenIn
+toOpenIn form
+  | isOpenIn form =
+    case form of
+      _OpenIn Sexp.:> sexp1 Sexp.:> sexp2 Sexp.:> Sexp.Nil ->
+        OpenIn sexp1 sexp2 |> Just
+      _ ->
+        Nothing
+  | otherwise =
+    Nothing
+
+fromOpenIn :: OpenIn -> Sexp.T
+fromOpenIn (OpenIn sexp1 sexp2) =
+  Sexp.list [Sexp.atom nameOpenIn, sexp1, sexp2]
+
+----------------------------------------
+-- Open
+----------------------------------------
+
+nameOpen :: NameSymbol.T
+nameOpen = "open"
+
+isOpen :: Sexp.T -> Bool
+isOpen (Sexp.Cons form _) = Sexp.isAtomNamed form nameOpen
+isOpen _ = False
+
+toOpen :: Sexp.T -> Maybe Open
+toOpen form
+  | isOpen form =
+    case form of
+      _Open Sexp.:> sexp1 Sexp.:> Sexp.Nil ->
+        Open sexp1 |> Just
+      _ ->
+        Nothing
+  | otherwise =
+    Nothing
+
+fromOpen :: Open -> Sexp.T
+fromOpen (Open sexp1) =
+  Sexp.list [Sexp.atom nameOpen, sexp1]
+
+----------------------------------------
+-- Declare
+----------------------------------------
+
+nameDeclare :: NameSymbol.T
+nameDeclare = "declare"
+
+isDeclare :: Sexp.T -> Bool
+isDeclare (Sexp.Cons form _) = Sexp.isAtomNamed form nameDeclare
+isDeclare _ = False
+
+toDeclare :: Sexp.T -> Maybe Declare
+toDeclare form
+  | isDeclare form =
+    case form of
+      _Declare Sexp.:> sexp1 Sexp.:> Sexp.Nil ->
+        Declare sexp1 |> Just
+      _ ->
+        Nothing
+  | otherwise =
+    Nothing
+
+fromDeclare :: Declare -> Sexp.T
+fromDeclare (Declare sexp1) =
+  Sexp.list [Sexp.atom nameDeclare, sexp1]
+
+----------------------------------------
+-- Declaim
+----------------------------------------
+
+nameDeclaim :: NameSymbol.T
+nameDeclaim = ":declaim"
+
+isDeclaim :: Sexp.T -> Bool
+isDeclaim (Sexp.Cons form _) = Sexp.isAtomNamed form nameDeclaim
+isDeclaim _ = False
+
+toDeclaim :: Sexp.T -> Maybe Declaim
+toDeclaim form
+  | isDeclaim form =
+    case form of
+      _Declaim Sexp.:> sexp1 Sexp.:> sexp2 Sexp.:> Sexp.Nil ->
+        Declaim sexp1 sexp2 |> Just
+      _ ->
+        Nothing
+  | otherwise =
+    Nothing
+
+fromDeclaim :: Declaim -> Sexp.T
+fromDeclaim (Declaim sexp1 sexp2) =
+  Sexp.list [Sexp.atom nameDeclaim, sexp1, sexp2]
+
+----------------------------------------
+-- DefModule
+----------------------------------------
+
+nameDefModule :: NameSymbol.T
+nameDefModule = ":defmodule"
+
+isDefModule :: Sexp.T -> Bool
+isDefModule (Sexp.Cons form _) = Sexp.isAtomNamed form nameDefModule
+isDefModule _ = False
+
+toDefModule :: Sexp.T -> Maybe DefModule
+toDefModule form
+  | isDefModule form =
+    case form of
+      _DefModule Sexp.:> sexp1 Sexp.:> sexp2 Sexp.:> sexp3 ->
```

Comment:
This _DefModule is a weird name. It refers to the name I think. Maybe prepending name to it? E.g. _nameDefModule?

Reply:
Do you want this to be true for all names or just this one?

Reply:
All names. I guess it'd be a small change as it's autogenerated

Reply:
Those changes are now in